central/signatureintegration/datastore/datastore_impl.go

```diff
@@ -149,6 +158,45 @@ func (d *datastoreImpl) verifyIntegrationIDDoesNotExist(ctx context.Context, id
 	return nil
 }
 
+func (d *datastoreImpl) verifyIntegrationIDIsNotInPolicy(ctx context.Context, id string) error {
+	// We want to avoid deleting a signature integration which is referenced by any policy. If that is the case,
+	// stop the deletion of the policy and return an appropriate error message to the user.
+	policies, err := d.policyStore.GetAllPolicies(ctx)
```

Comment:
What if the principal making a request has no permission to read policies? Looks like you will happily remove the integration in this case. A usual approach is to use different (admin) context here.